deno/lib/stage2.ts

```diff
@@ -14,16 +14,26 @@ const getFunctionReference = (basePath: string, func: InputFunction, index: numb
   return {
     exportLine,
     importLine: `import ${importName} from "${url}";`,
+    metadata: {
+      url,
+    },
+    name: func.name,
   }
 }
 
-const getStage2Entry = (basePath: string, functions: InputFunction[]) => {
+export const getStage2Entry = (basePath: string, functions: InputFunction[]) => {
   const lines = functions.map((func, index) => getFunctionReference(basePath, func, index))
   const importLines = lines.map(({ importLine }) => importLine).join('\n')
   const exportLines = lines.map(({ exportLine }) => exportLine).join(', ')
-  const exportDeclaration = `export const functions = {${exportLines}};`
+  const metadataLines = lines.map(({ metadata, name }) => {
+    const metadataString = JSON.stringify(metadata)
 
-  return [importLines, exportDeclaration].join('\n\n')
+    return `"${name}":${metadataString}`
+  })
+  const functionsExport = `export const functions = {${exportLines}};`
+  const metadataExport = `export const metadata = {${metadataLines}};`
```

Comment:
this scares me a bit. why are you doing this line by line instead of just calling `JSON.stringify` on a whole object?
also: did you forget a `.join(",")`?

Reply:
Great point! Updated in https://github.com/netlify/edge-bundler/pull/122/commits/71bed6300250f5e4fe97296b2602535680b9dabf.